app/routes/session.py

```diff
@@ -129,17 +134,41 @@ def login() -> Response:
 
     cookie_session["language_code"] = metadata.language_code
 
-    # Type ignore: survey_id and either ru_ref or qid are required for schemas that use supplementary data
-    if dataset_id := metadata["sds_dataset_id"]:
-        get_supplementary_data(
-            dataset_id=dataset_id,
-            unit_id=metadata["ru_ref"] or metadata["qid"],  # type: ignore
-            survey_id=metadata["survey_id"],  # type: ignore
-        )
-
     return redirect(url_for("questionnaire.get_questionnaire"))
 
 
+def _set_questionnaire_supplementary_data(
+    questionnaire_store: QuestionnaireStore, metadata: MetadataProxy
+) -> None:
+    """
+    If the survey metadata has an sds dataset id, and it either doesn't match what it stored, or there is no stored supplementary data
+    then fetch it and add it to the store
+    """
+    existing_sds_dataset_id = (
+        questionnaire_store.metadata.survey_metadata["sds_dataset_id"]
+        if questionnaire_store.metadata and questionnaire_store.metadata.survey_metadata
+        else None
+    )
+
+    if not (new_sds_dataset_id := metadata["sds_dataset_id"]):
+        if existing_sds_dataset_id:
+            # remove what is already there
+            questionnaire_store.set_supplementary_data({})
+        return
```

Comment:
As mentioned in the other comment, this could result in orphaned list items and answers.